RSHighscores.body.php

```diff
@@ -69,6 +69,42 @@ private static function retrieveHiscores( $hs, $player ) {
 	}
 
 	/**
+	 * Lookup hiscores data from object cache before retrieving from the site.
+	 *
+	 * @param string $hs Which hiscores API to retrieve from.
+	 * @param string $player Player's display name.
+	 * @return string Raw hiscores data
+	 */
+	private static function lookupHiscores( $hs, $player ) {
+		// Instance of the object cache to retrieve the hiscores data from.
+		$objCache = wfGetCache( CACHE_ANYTHING );
+
+		// Try to retrieve the hiscores data from the object cache.
+		$data = $objCache->get( 'rshiscores-' . $player . '-' . $hs );
+
+		// Couldn't find in the object cache, so retrieve from the site.
+		if ( $data === false ) {
+			// If not blocked from too many requests or technical issues, then lookup.
+			if ( $objCache->get( 'rshiscores-blocked') === false ) {
+				$data = self::retrieveHiscores( $hs, $player );
+
+				// Request failed, so no requests for 15 min.
+				if ( $data == 'C28' ) {
+					$objCache->set( 'rshiscores-blocked', true, 60 * 15 );
+				}
+
+				// Cache the new results.
+				$objCache->set( 'rshiscores-' . $player . '-' . $hs, $data, 60 );
+			} else {
+				// This or a previous request failed, so hold off further requests for now.
+				$data = 'I';
```

Comment:
If a previous request failed, this should this check if there's a C or D error that can be returned instead?